source/models/index.py

```diff
@@ -0,0 +1,49 @@
+from abc import ABC, abstractmethod
+import os
+from outputs.exceptions import FileNotFound
+from models.table import Table
+
+class index(ABC):
+    def __init__(self, database_name, table_name):
+        self.indices_path = database_name + "\\" + table_name + "\\indices" 
+        self.database_name = database_name
+        self.table_name = table_name
+
+    def getIndicesNames(self):
+        return os.listdir(self.indices_path)
+
+    def getprimary_keys(self, indexName, index):
+        file_path = self.indices_path + "\\" + indexName + "\\" + index + ".txt"
+        if not os.path.isfile(file_path):
+            raise FileNotFound(message = "this index has not been saved to the database yet")
+        with open(file_path,"r")as primary_keys:
+            primary_keys = primary_keys.read().split(",")
+        return primary_keys
+    
+    def addIndices(self, row_obj):
+        table = Table(self.database_name, self.table_name)
```

Comment:
This loop should be in class table. 
in. class table, you call add_index

Reply:
the metadata class is the class that calls the serialize function of the index so I wrote this for loop in the metadata class in the new version of the index.py